eng/mgmt/automation/generate_data.py

```diff
@@ -64,11 +64,7 @@ def sdk_automation_cadl(config: dict) -> List[dict]:
                 os.chdir(cadl_dir)
                 try:
                     # install dependencies
-                    subprocess.run('npm install "@azure-tools/cadl-autorest@0.24.0" "@cadl-lang/eslint-config-cadl@0.5.0" "@cadl-lang/eslint-plugin@0.38.0" "@cadl-lang/library-linter@0.38.0"', shell=True, check=True)
-
-                    # install latest @azure-tools/cadl-java
-                    subprocess.run('npm install "@azure-tools/cadl-java@0.2.2"', shell=True, check=True)
-
+                    subprocess.run('npm install "@azure-tools/cadl-autorest@0.24.0" "@azure-tools/cadl-azure-core@0.24.0" "@cadl-lang/compiler@0.38.5" "@cadl-lang/eslint-config-cadl@0.5.0" "@cadl-lang/eslint-plugin@0.38.0" "@cadl-lang/library-linter@0.38.0" "@azure-tools/cadl-java@0.2.2"', shell=True, check=True)
```

Comment:
@dw511214992 why aren't we moving to the common model of using a package.json file for the java emitter and installing the packages from that.

Reply:
We had one here https://github.com/Azure/azure-sdk-for-java/blob/main/eng/emitter-package.json by Srikanta. But we might need some discussion and some tuning there (with the cadl-location.yaml), as you can see it only contains cadl-java

Reply:
I'd like to see us get to the common model sooner rather than later, but I'll leave it up to you when you make that transition.

Reply:
@weidongxu-microsoft  do you want to get to the common model now? If yes, I will close this PR. thanks

Reply:
Thanks Wes and Wei.

I've added an issue to outline what's remaining to be done. https://github.com/Azure/azure-sdk-for-java/issues/33346

Let me know if this meets your expectation.